Smt/Commands.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2021-2022 by the authors listed in the file AUTHORS and their
+institutional affiliations. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Abdalrhman Mohamed, Tomaz Gomes Mascarenhas, Wojciech Nawrocki
+-/
+
+import Smt.Solver
+import Smt.Term
+
+namespace Smt
+
+/-- An SMT-LIBv2 command that we can emit. -/
+inductive Command where
```

Comment:
I found it useful to have an explicit datatype for commands rather than weaving `Solver` state through. With this, we can store `Command`s in the dependency graph and then just send them to `Solver` during the DFS. I think this should also work with a `SolverM` which immediately sends the commands over to a process.

Note I only have `defineFun` and `defineConst = defineFun (ps := [])`.